ckanext/vocabulary_services/logic/action/create.py

```diff
@@ -77,9 +79,11 @@ def vocabulary_service_term_upsert(context, data_dict):
                 if (existing_term.label == label) and (existing_term.uri != uri):
                     # If label is the same but uri is different, let's create them.
                     vocabulary_service_term_create(context, data_dict)
-                elif (existing_term.label != label) or (existing_term.uri == uri):
+                elif (existing_term.label != label or existing_term.definition != definition) and existing_term.uri == uri:
```

Comment:
@salsa-nathan  I updated the logic here to match the comment 'Update the term label if the URI is the same and label different.
Not sure if this was a bug or intended but I updated the logic from `or (existing_term.uri == uri)` to `and (existing_term.uri == uri)`

Reply:
@MarkCalvert - yes I think the change you made makes sense